Source/WarpXIO.cpp

```diff
@@ -797,12 +808,22 @@ WarpX::WritePlotFile () const
             }
 	    dcomp += 3;
 
-	    PackPlotDataPtrs(srcmf, Efield_aux[lev]);
-	    amrex::average_edge_to_cellcenter(*mf[lev], dcomp, srcmf);
+            // E
+            if (do_nodal)
+            {
+                amrex::average_node_to_cellcenter(*mf[lev], dcomp  , *Efield_aux[lev][0], 0, 1);
+                amrex::average_node_to_cellcenter(*mf[lev], dcomp+1, *Efield_aux[lev][1], 0, 1);
+                amrex::average_node_to_cellcenter(*mf[lev], dcomp+2, *Efield_aux[lev][2], 0, 1);
```

Comment:
@WeiqunZhang 
Could you explain why we can use `PackPlotDataPtrs` in the case of a staggered grid, but not in the case of a nodal grid?

Reply:
`amrex::average_node_to_cellcenter` and `amrex::average_edge_to_cellcenter` have different interface.   The node version takes two MultiFabs, whereas the edge version takes an Array of edge MultiFabs and a cell-centered MultiFab.